docs/spec/spec.md

```diff
@@ -0,0 +1,388 @@
+_Updated_: 2024/02/28 \
+_Edition_: Swan Lake
+
+## Introduction
+
+This is the specification for the Redis connector library of the [Ballerina language](https://ballerina.io),
+which provides a comprehensive set of operations to interact with Redis servers using Ballerina.
+
+The Redis library specification undergoes continuous refinement, and any updates are documented and released under
+relevant GitHub tags.
+
+If you have any feedback or suggestions about the library, start a discussion via
+a [GitHub issue](https://github.com/ballerina-platform/ballerina-library/issues)
+or in the [Discord server](https://discord.gg/ballerinalang).
+Based on the outcome of the discussion, the specification and implementation can be updated. Community feedback is
+always welcome.
+
+Any accepted proposal, which affects the specification is stored under `/docs/proposals`. Proposals under discussion can
+be found with the label `type/proposal` in GitHub.
+
+The conforming implementation of the specification is released to Ballerina Central. Any deviation from the
+specification is considered a bug.
+
+## Contents
+
+1. [Overview](#1-overview)
+2. [Client](#2-client)  
+    2.1. [Configurations](#21-configurations)  
+    2.2. [Initialization](#22-initialization)  
+    2.3. [Secure communication](#23-secure-communication)  
+    2.4. [Cluster connections](#24-cluster-connections)  
+    2.5. [Connection pooling](#25-connection-pooling)  
+    2.6. [Closing the connection](#26-closing-the-connection)
+3. [Supported Operations](#3-supported-operations)  
+   3.1. [Hash Operations](#31-hash-operations)  
+   3.2. [Key Operations](#32-key-operations)  
+   3.3. [List Operations](#33-list-operations)  
+   3.4. [Set Operations](#34-set-operations)  
+   3.5. [Sorted Set Operations](#35-sorted-set-operations)  
+   3.6. [String Operations](#36-string-operations)  
+   3.7. [Cluster Operations](#37-cluster-operations)
+
+# 1. Overview
+
+Redis (REmote DIctionary Server) is an open-source, in-memory data structure store that can be used as a database,
+cache, and message broker.
+It supports various data structures such as strings, hashes, lists, sets, and more. Redis has its own protocol for
+communication between clients and the server.
+
+The Ballerina Redis client library supports a multitude of Redis operations related to different data structures, such
+as key-value operations, list operations, and more. The client also supports connecting to Redis clusters, which allows
+working with multiple Redis nodes using a single client. It also comes with the support for various connection features
+such as connection pooling, which allows reusing connections to the Redis server, and SSL/TLS encryption for secure
+communication with the Redis server.
+
+# 2. Client
+
+The Redis `Client` represents a connection to the Redis server. It maintains a pool of connections throughout its
+lifetime.
+
+## 2.1. Configurations
+
+### 2.1.1. Connection configurations
```

Comment:
Do we need to add this as titles ? If so, lets add them to the TOC.

Reply:
Updated the ToC with https://github.com/ballerina-platform/module-ballerinax-redis/pull/219/commits/cbf0dcec316b290dcffe012437d56e35d2e4c97c